### ISSUE TITLE
Refactor regular text extraction into dedicated module

### ISSUE THREAD

**Issue by stefan6419846** [py-pdf/pypdf#3010]
At the moment, the regular (non-layout mode) text extraction is mostly implemented in a method inside the `PageObject` class directly. This makes it hard to keep the complexity low, requires `nonglobal` variables and bloats the `PageObject` more and more on every new error handler.

For this reason, we should look into moving the actual internal implementation to a dedicated module/class to allow for easier refactoring and reduce file/module complexity.

**Comment by MartinThoma** [comment]
Maybe creating a TextExtraction class would help:

* We could replace the `nonlocal` keywords by object attributes
* We could get rid of the nested functions
* We could introduce methods that handle operations (_handle_operation_save_graphics_state, _handle_operations_restore_graphics_state, _handle_operation_horizontal_text_scaling, ...)
    * That would allow us to have a dictionary mapping the operator to the operation which could reduce the complexity of process_operation quite a bit

What do you think?

**Comment by stefan6419846** [comment]
Yes, this is similar to what I have been thinking. Regarding the different operations, this probably depends on whether their API signatures can be unified well enough, but the easier to read and test, the better.
